src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -74,5 +75,29 @@ internal static bool IsJObject(Type type)
 
             return null;
         }
+
+        public static bool IsAsync(MethodInfo methodInfo)
+        {
+            if (methodInfo == null)
+            {
+                throw new ArgumentNullException(nameof(methodInfo));
+            }
+
+            var stateMachineAttribute = methodInfo.GetCustomAttribute<AsyncStateMachineAttribute>();
```

Comment:
I think the documentation on these attributes is clear enough. It says "Indicates whether a method is marked with either the Async (Visual Basic) or async (C# Reference) modifier."